src/app/app.component.ts

```diff
@@ -81,8 +87,26 @@ export class MyApp {
   openPage(page) {
     // Reset the content nav to have just this page
     // we wouldn't want the back button to show in this scenario
-    this.menuCtrl.close().then(() => {  
-      this.nav.setRoot(page.component, { 'linkParam' : page.param });
+    this.menuCtrl.close().then(() => {
+      let changePageAllowed: boolean = true;
+      if(page.component === PushMessagesPage) {
+        if(!PushService.registered) {
+          changePageAllowed = false;
+          this.alert.showAlert(constants.pushAlertMessage, constants.featureNotConfigured, 
+            constants.alertButtons, constants.showDocs, constants.pushDocsUrl);
+        } 
+      } else if (page.component === AuthPage) {
+        const configKeys: string[] = Object.keys(this.auth.getConfig()).toString().split(',');
```

Comment:
Is this not the same as doing `const configKeys = Object.keys(this.auth.getConfig());`. 

* The TypeScript compiler will determine it's an array of strings
* We're converting an array to a string and then back to an array again, from what I can see